src/shims/io.rs

```diff
@@ -151,10 +173,38 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         )
     }
 
+    fn write(
+        &mut self,
+        fd_op: OpTy<'tcx, Tag>,
+        buf_op: OpTy<'tcx, Tag>,
+        count_op: OpTy<'tcx, Tag>,
+    ) -> InterpResult<'tcx, i64> {
+        let this = self.eval_context_mut();
+
+        if !this.machine.communicate {
+            throw_unsup_format!("`write` not available when isolation is enabled")
+        }
+
+        let tcx = &{ this.tcx.tcx };
+
+        let fd = this.read_scalar(fd_op)?.to_i32()?;
+        let buf = this.force_ptr(this.read_scalar(buf_op)?.not_undef()?)?;
+        let count = this.read_scalar(count_op)?.to_usize(&*this.tcx)?;
+
+        // `to_vec` is needed to avoid borrowing issues when writing to the file.
```

Comment:
Won't it work if you move this into the closure just like in read?

Reply:
yes, but I'm afraid of removing the file from the `FileHandler` and then failing without adding it back

Reply:
Read has the same problem. You can fix it with a bunch of and_then calls instead of `?` invocations

Reply:
Oh damn. I'll do it